superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -34,17 +37,103 @@ export const FormFieldOrder = [
   'username',
   'password',
   'database_name',
+  'credentials_info',
 ];
 
 interface FieldPropTypes {
   required: boolean;
   changeMethods: { onParametersChange: (value: any) => string } & {
     onChange: (value: any) => string;
-  };
+  } & { onParametersUploadFileChange: (value: any) => string };
   validationErrors: JsonObject | null;
   getValidation: () => void;
 }
 
+const credentialsInfo = ({
+  required,
+  changeMethods,
+  getValidation,
+  validationErrors,
+}: FieldPropTypes) => {
+  const [uploadOption, setUploadOption] = useState<string>('upload');
+  const [fileToUpload, setFileToUpload] = useState<string>(null);
+  return (
+    <CredentialInfoForm>
+      <>
+        <label className="label-select">
+          How do you want to enter service account credentials?
+        </label>
+        <Select
+          defaultValue={'file'}
+          style={{ width: '100%' }}
+          onChange={option => setUploadOption(option)}
+        >
+          <Select value="file">Upload JSON file</Select>
+          <Select value="paste">Copy and Paste JSON credentials</Select>
+        </Select>
+        {uploadOption === 'paste' ? (
+          <div className="input-container" onChange={changeMethods.onChange}>
+            <label className="label-select">Service Account</label>
+            <textarea className="input-form" name="encrypted_extra" />
+            <label className="label-paste">
+              Copy and paste the entire service account .json file here
+            </label>
+          </div>
+        ) : (
+          <div className="input-container">
+            <label className="label-select">Upload Credentials</label>
+            {!fileToUpload && (
+              <>
+                <Button
+                  className="input-upload-btn"
+                  onClick={() =>
+                    document.getElementById('selectedFile').click()
+                  }
+                >
+                  Choose File
+                </Button>
+              </>
+            )}
+            {fileToUpload && (
+              <div className="input-upload-current">
+                {fileToUpload}
+                <DeleteFilled
+                  onClick={() => {
+                    setFileToUpload(null);
+                    changeMethods.onParametersUploadFileChange({
+                      target: {
+                        name: 'encrypted_extra',
+                        value: '',
+                      },
+                    });
+                  }}
+                />
+              </div>
+            )}
+
+            <input
+              id="selectedFile"
+              className="input-upload"
+              type="file"
+              onChange={async event => {
```

Comment:
nit, but this event  function is a little long. I'd suggest putting it into a separate function.